compiler/rustc_typeck/src/check/method/confirm.rs

```diff
@@ -176,15 +188,32 @@ impl<'a, 'tcx> ConfirmContext<'a, 'tcx> {
                 adjustments.push(Adjustment { kind: Adjust::Pointer(PointerCast::Unsize), target });
             }
         } else {
-            // No unsizing should be performed without autoref (at
-            // least during method dispach). This is because we
-            // currently only unsize `[T;N]` to `[T]`, and naturally
-            // that must occur being a reference.
+            // No unsizing should be performed without autoref (at least during
+            // method dispach). This is because we currently only unsize `[T;N]`
+            // to `[T]`, and naturally that must occur being a reference.
             assert!(pick.unsize.is_none());
+
+            if pick.to_const_ptr {
```

Comment:
This could use a comment -- what case is it handling?

Reply:
`to_const_ptr` is only set when the receiver object has type `*mut` and the method's self is `*const`. Quoting the documentation:

```rust
    /// Only valid when the receiver type is a `*mut`: coerce the receiver to `*const`. If this it
    /// `true` then `autoref` should be `None`.
    pub to_const_ptr: bool,
```

I will add a comment here.